src/python/pants/reporting/reporting_server.py

```diff
@@ -307,15 +315,17 @@ def _serve_file(self, abspath, params):
     """
     relpath = os.path.relpath(abspath, self._root)
     breadcrumbs = self._create_breadcrumbs(relpath)
-    link_path = urlunparse([None, None, relpath, None, urlencode(params), None])
+    link_path = urlunparse(['', '', relpath, '', urlencode(params), ''])
     args = self._default_template_args('file.html')
     args.update({'root_parent': os.path.dirname(self._root),
                  'breadcrumbs': breadcrumbs,
                  'link_path': link_path})
-    self._send_content(self._renderer.render_name('base.html', args), 'text/html')
+    content = self._renderer.render_name('base.html', args).encode("utf-8")
+    self._send_content(content, 'text/html')
 
   def _send_content(self, content, content_type, code=200):
     """Send content to client."""
+    assert isinstance(content, bytes)
```

Comment:
This assertion is added so that Python 2 behaves like Python 3 does.

Note we do not coerce the content here, because we don't know what the encoding should be, e.g. if it's an asset that doesn't use UTF-8. So, we require the caller to encode its content beforehand.